portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login-status")
+public class LoginServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    boolean loggedIn = userService.isUserLoggedIn();
+    String url = loggedIn ? userService.createLogoutURL("/") : userService.createLoginURL("/");
+
+    response.setContentType("application/json;");
+    response.getWriter().println(toJson(loggedIn, url));
+  }
+
+  /* Return json string that specifies if the user is logged in and
+  provides a login/logout url */
+  public String toJson(boolean loggedIn, String url) {
```

Comment:
Two comments:
 - I think you can simply pass `true` and `false` to the json (without the quotation marks around them) and they should be parsed as booleans. Then in JS you can simply check json.logged without having to compare it to the strings "true" and "false".
 - This method is short but hard to read. Consider using a small holder class - something like a struct, e.g. 
```
private class UserStatus {
  bool logged;
  String url;
}
```
 if you feed that to Gson it will automatically format those for you.